lib/recipe.rb

```diff
@@ -0,0 +1,41 @@
+class Recipe
+  attr_reader :label, :image, :uri, :url, :ingredient_list, :health_labels, :source
+  def initialize(label, image, uri, url, ingredient_list, health_labels, source)
+    valid_attribute?(label)
+    valid_attribute?(image)
+    valid_attribute?(uri)
```

Comment:
I would say you're over-validating here. You should only validate attributes that will cause your site to break if they're absent.

As an example of this, when I searched for "tacos" one of the results had no health labels, which produced an error here.